libraries/Microsoft.Bot.Builder/BotFrameworkStreamingExtensionsAdapter.cs

```diff
@@ -0,0 +1,695 @@
+        {
+            MiddlewareSet.Use(middleware);
+            return this;
+        }
+
+        /// <summary>
+        /// Overload for processing activities when given the activity a json string representation of a request body.
+        /// Creates a turn context and runs the middleware pipeline for an incoming activity.
+        /// Throws <see cref="ArgumentNullException"/> on null arguments.
+        /// </summary>
+        /// <param name="body">The json string to deserialize into an <see cref="Activity"/>.</param>
+        /// <param name="streams">A set of streams associated with but not attached to the <see cref="Activity"/>.</param>
+        /// <param name="callback">The code to run at the end of the adapter's middleware pipeline.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute. If the activity type
+        /// was 'Invoke' and the corresponding key (channelId + activityId) was found
+        /// then an InvokeResponse is returned, otherwise null is returned.</returns> /// <returns>a <see cref="Task"/> that will resolve a <see cref="InvokeResponse"/> in response to the activity.</returns>
+        /// <remarks>Call this method to reactively send a message to a conversation.
+        /// If the task completes successfully, then if the activity's <see cref="Activity.Type"/>
+        /// is <see cref="ActivityTypes.Invoke"/> and the corresponding key
+        /// (<see cref="Activity.ChannelId"/> + <see cref="Activity.Id"/>) is found
+        /// then an <see cref="InvokeResponse"/> is returned, otherwise null is returned.
+        /// <para>This method registers the following services for the turn.<list type="bullet">
+        public async Task<InvokeResponse> ProcessActivityAsync(string body, List<IContentStream> streams, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(body))
+            {
+                throw new ArgumentNullException(nameof(body));
+            }
+
+            if (streams == null)
+            {
+                throw new ArgumentNullException(nameof(streams));
+            }
+
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            if (cancellationToken == null)
+            {
+                throw new ArgumentNullException(nameof(cancellationToken));
+            }
+
+            var activity = JsonConvert.DeserializeObject<Activity>(body, SerializationSettings.DefaultDeserializationSettings);
+
+            if (streams.Count > 1)
+            {
+                var streamAttachments = new List<Attachment>();
+                for (var i = 1; i < streams.Count; i++)
+                {
```

Comment:
Please add a detailed comment here. I don't know/understand why there are streams and for loops here.